### ISSUE TITLE
[🐛BUG] SGD doesn't learn

### ISSUE THREAD

**Issue by macafrap** [RUCAIBox/RecBole#1778]
**Describe the bug**
When using SGD as a learner, the model doesn't learn in 300 epochs, but metric values and train loss stay (almost) exactly the same. When using a seed, the metric/loss values stay the same regardless of specified learning rate. This is the case for at least the BPR and NeuMF model (and I expect all other models too). 

**To Reproduce**
I run the python file run.py:
"run_recbole(model='BPR', dataset='ml-100k')" 
by calling "python run.py --learner='sdg' "

**Expected behavior**
I expect the loss to change over time, and the metrics to improve over time.

**Comment by Sherry-XLL** [comment]
Hello @macafrap, thanks for your attention to RecBole!

For the optimizer of stochastic gradient descent (SGD), we need to add the parameter `momentum` to promote the parameter optimization, and the learning rate needs to be set to a relatively large value.

If you want to use the SGD learner for ml-100k dataset, please change the definition of `optimizer` in `recbole.trainer.trainer.py` as follows.
The original code:
https://github.com/RUCAIBox/RecBole/blob/a757b7c31bfa407c82847b4a06c8580fa1a6b45e/recbole/trainer/trainer.py#L179

The modified code:
```
optimizer = optim.SGD(params, lr=learning_rate, weight_decay=weight_decay, momentum=0.9)
```

You can change the value of `momentum` if needed. Then, you can set the `learning_rate` to 0.5, and you can get the following output:
```
$ python run_recbole.py --learner='sgd' --learning_rate=0.5
... ...
24 May 14:57    INFO  epoch 18 training [time: 0.18s, train loss: 27.3451]
Evaluate   : 100%|█████████████████████| 472/472 [00:00<00:00, 1073.17it/s, GPU RAM: 0.03 G/39.59 G]
24 May 14:57    INFO  epoch 18 evaluating [time: 0.45s, valid_score: 0.157700]
24 May 14:57    INFO  valid result: 
recall@10 : 0.0425    mrr@10 : 0.1577    ndcg@10 : 0.0691    hit@10 : 0.3393    precision@10 : 0.0592
24 May 14:57    INFO  Saving current: saved/BPR-May-24-2023_14-57-06.pth
Train    19: 100%|████████████████████████| 40/40 [00:00<00:00, 222.78it/s, GPU RAM: 0.03 G/39.59 G]
24 May 14:57    INFO  epoch 19 training [time: 0.18s, train loss: 27.2458]
Evaluate   : 100%|█████████████████████| 472/472 [00:00<00:00, 1063.17it/s, GPU RAM: 0.03 G/39.59 G]
24 May 14:57    INFO  epoch 19 evaluating [time: 0.45s, valid_score: 0.174400]
24 May 14:57    INFO  valid result: 
recall@10 : 0.0517    mrr@10 : 0.1744    ndcg@10 : 0.0794    hit@10 : 0.3754    precision@10 : 0.0677
24 May 14:57    INFO  Saving current: saved/BPR-May-24-2023_14-57-06.pth
Train    20: 100%|████████████████████████| 40/40 [00:00<00:00, 224.21it/s, GPU RAM: 0.03 G/39.59 G]
24 May 14:57    INFO  epoch 20 training [time: 0.18s, train loss: 27.1105]
Evaluate   : 100%|█████████████████████| 472/472 [00:00<00:00, 1069.10it/s, GPU RAM: 0.03 G/39.59 G]
24 May 14:57    INFO  epoch 20 evaluating [time: 0.45s, valid_score: 0.188800]
24 May 14:57    INFO  valid result: 
recall@10 : 0.0567    mrr@10 : 0.1888    ndcg@10 : 0.0871    hit@10 : 0.3955    precision@10 : 0.0736
24 May 14:57    INFO  Saving current: saved/BPR-May-24-2023_14-57-06.pth
... ...
```

By the way, we recommend `Adam` as the learner since it performs more stably than `SGD`.
